### PR TITLE
Rename module to builtin.module

### PR DESCRIPTION
**Pull request by math-fehr**
In MLIR, `module` real name is `builtin.module`.
It was named `module` in xDSL, but this mess up with the mlir_converter, and the MLIR printer.
This should fix it, but this will be a breaking change.

### CODE REVIEW

**Comment by math-fehr** [comment]
I also added a changelog file, so we can now start tracking changes, most importantly breaking changes

**Review by Dinistro — APPROVED** [review]
LGTM.
This was in fact a breaking change in MLIR at some point in the last 12 months. When we started with this work, the module omitted the `builtin` prefix.

> I also added a changelog file, so we can now start tracking changes, most importantly breaking changes

That's a great idea. Back when I was a web developer, we used a tool to generate them from the commit messages. Not sure if this would be nice to have here as well.

File: tests/filecheck/mlir-conversion/llvm_test.xdsl [hunk 1]
```diff
@@ -1,26 +1,26 @@
 // RUN: xdsl-opt -t mlir %s | filecheck %s
 
 
-module() {
+builtin.module() {
   // Type tests
   func.func() ["sym_name" = "struct_to_struct", "function_type" = !fun<[!llvm.struct<"", [!i32]>], [!llvm.struct<"", [!i32]>]>, "sym_visibility" = "private"] {
-  // CHECK: func.func private @struct_to_struct(%arg0: !llvm.struct<(i32)>) -> !llvm.struct<(i32)> {
+  // CHECK: func private @struct_to_struct(%arg0: !llvm.struct<(i32)>) -> !llvm.struct<(i32)> {
```
Comment (webmiche) [inline tests/filecheck/mlir-conversion/llvm_test.xdsl:7]: This change is unrelated, right? On which LLVM version are you? The commit in the README?

**Review by webmiche — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Changelog
```
Comment (georgebisbas) [inline CHANGELOG.md:1]: This is what https://github.com/xdslproject/xdsl/blob/main/.github/release-drafter.yml
is meant to automatically work on.
PLease add a tag/label to your PR and every PR from now on.

**Review by georgebisbas — CHANGES_REQUESTED** [review]

File: tests/filecheck/mlir-conversion/llvm_test.xdsl (hunk 1 above)
Comment (math-fehr) [reply tests/filecheck/mlir-conversion/llvm_test.xdsl:7]: I had to write this to make the test pass again. I was on the commit in the readme (7499...)

File: CHANGELOG.md (hunk 2 above)
Comment (math-fehr) [reply CHANGELOG.md]: Oh okay, I now understand, thanks!

**Comment by math-fehr** [comment]
This now depends on #163

File: tests/filecheck/mlir-conversion/llvm_test.xdsl (hunk 1 above)
Comment (math-fehr) [reply tests/filecheck/mlir-conversion/llvm_test.xdsl:7]: Moved it to a separate commit!
